### ISSUE TITLE
Find and prevent system proxy leaks

### ISSUE THREAD

**Issue by diracdeltas** [brave/browser-laptop#2103]
Node does not generally use system proxy settings, so there are some proxy leaks in Brave. ex: https://github.com/brave/browser-laptop/issues/770#issuecomment-223782475

**Comment by diracdeltas** [comment]
known leaks:
* app/updater.js in getting brave updates
* app/datafile.js for downloading updates to ABP, tracking protection, https everywhere, etc. files

**Comment by diracdeltas** [comment]
@aekeus I'm thinking that on windows, if system proxy settings are detected, we should just download the update directly instead of going through the auto-updater. thoughts?

**Comment by aekeus** [comment]
On Windows it is a three step process.

  * Check with our metadata server for pending upgrade
  * Download RELEASES file that points to update binary
  * Download and apply update binary

Squirrel handles step 2 and 3 directly, we control only step 1

**Comment by diracdeltas** [comment]
Step 1 should be properly proxied now. I meant that if proxy settings are detected, we should do step 2 and 3 ourselves instead of going through Squirrel

**Comment by aekeus** [comment]
Right, that makes sense. I am not sure of the effort required to implement 2 and 3 ourselves.
